tests/tests.py

```diff
@@ -407,6 +449,158 @@ def test_add_remove_attachment(self):
 
 
 
+class EntryHistoryTests3(KDBX3Tests):
+
+    #---------- History -----------
+
+    def test_find_history_entries(self):
+        '''run some tests on entries created by pykeepass'''
+        prefix = 'TFE_'
+        postfix = '_tmpA'
+        changed = 'tfe_changed_'
+        # create a new copy of the file to ensure the original file remains unchanged
+        a = self.kp.filename.split('.')
+        fn = '.'.join(('.'.join(a[:-1]) + postfix, a[-1]))
+        keyfile = os.path.join(base_dir, self.keyfile)
+        self.kp.save(filename=fn)
+
+        # create some new entries to have clean start
+        with PyKeePass(fn, password=self.password, keyfile=keyfile) as kp2:
+            e1 = kp2.add_entry(kp2.root_group,
+                prefix + 'title',
+                prefix + 'user',
+                prefix + 'pass'
+            )
+            g1 = kp2.add_group(kp2.root_group, prefix + 'group')
+            e2 = kp2.add_entry(g1,
+                prefix + 'title',
+                prefix + 'user',
+                prefix + 'pass'
+            )
+            g2 = kp2.add_group(g1, prefix + 'sub_group')
+            e2 = kp2.add_entry(g2,
+                prefix + 'title',
+                prefix + 'user',
+                prefix + 'pass'
+            )
+            kp2.save()
+
+        # no history tests
+        with PyKeePass(fn, password=self.password, keyfile=keyfile) as kp2:
+            res1 = kp2.find_entries_by_title(prefix + 'title')
+            self.assertEqual(len(res1), 3)
+            for entry in res1:
+                self.assertFalse(entry.is_a_history_entry)
+                hist = entry.history
+                self.assertIsInstance(hist, list)
+                self.assertEqual(len(hist), 0)
+
+            res2 = kp2.find_entries_by_title(prefix + 'title', history=True)
+            self.assertEqual(len(res2), 3)
+
+            # create history
+            for entry in res1:
+                entry.save_history()
+            kp2.save()
+
+        # first history tests
+        with PyKeePass(fn, password=self.password, keyfile=keyfile) as kp2:
+            # we should not find any history items
+            res1 = kp2.find_entries_by_title(prefix + 'title')
+            self.assertEqual(len(res1), 3)
+            for entry in res1:
+                self.assertFalse(entry.is_a_history_entry)
+                hist = entry.history
+                self.assertEqual(len(hist), 1)
+                for item in hist:
+                    self.assertTrue(item.is_a_history_entry)
+                    self.assertEqual(item.group, entry.group)
+                    self.assertEqual(item.path, '[History of: {}]'.format(entry.title))
+                    # check if everthing was copied as expected
```

Comment:
Under the hood, `save_history()` just duplicates the Entry and appends it to history, so it's probably not necessary to test every entry attribute here.  We'd just be writing tests for lxml's `.append()` function.